salt/modules/state.py

```diff
@@ -2133,6 +2133,34 @@ def pkg(pkg_path,
     return ret
 
 
+def _cleanup_debug_ssh_pkgs(pkg_dir):
+    '''
+    Remove previously uploaded salt-ssh state packages.
+    Keep the most recent 10 packages for debugging purposes.
+    '''
+    pkgs = []
+    for s in os.listdir(pkg_dir):
+        if not (s.startswith('debug_salt_state') and s.endswith('.tgz')):
+            continue
+        pkg_path = os.path.join(pkg_dir, s)
+        pkgs.append((pkg_path, os.stat(pkg_path).st_mtime))
+    n_to_keep = 10
+    for pkg_path, _ in sorted(pkgs, key=lambda pkg: pkg[1])[:-n_to_keep]:
+        os.unlink(pkg_path)
+
+
+def pkg_ssh(pkg_path, *args, **kwargs):
```

Comment:
Can you add some documentation to this new function?

Reply:
Ok, will do.

Reply:
I realized that I don't, strictly speaking, need a new function here.

So I introduced a new parameter remove_pkg for the existing pkg(), which is False by default. salt-ssh sets it to True, and this triggers the new behavior.